src/LineSegment3d/Extra.elm

```diff
@@ -0,0 +1,391 @@
+                { param0 = 0
+                , param1 = getClampedRoot cfg.c cfg.g00 cfg.g01
+                }
+
+            else if cfg.edge1 == 1 then
+                { param0 = 1
+                , param1 = getClampedRoot cfg.c cfg.g10 cfg.g11
+                }
+
+            else
+                { param0 = x1
+                , param1 = y1
+                }
+
+        else
+            let
+                z =
+                    min (max (h0 / (h0 - h1)) 0) 1
+
+                omz =
+                    1 - z
+            in
+            { param0 = omz * x0 + z * x1
+            , param1 = omz * y0 + z * y1
+            }
+
+
+midIfOutOfBounds : Float -> Float
+midIfOutOfBounds n =
+    if n < 0 || n > 1 then
+        0.5
+
+    else
+        n
+
+
+computeIntersection :
+    { sValue0 : Float
+    , sValue1 : Float
+    , classify0 : Float
+    , classify1 : Float
+    , b : Float
+    , f00 : Float
+    , f10 : Float
+    }
+    ->
+        { edge0 : Float
+        , edge1 : Float
+        , end00 : Float
+        , end01 : Float
+        , end10 : Float
+        , end11 : Float
+        }
+computeIntersection cfg =
+    if cfg.classify0 < 0 then
+        let
+            ( edge1, end10, end11 ) =
+                if cfg.classify1 == 0 then
+                    ( 3, cfg.sValue1, 1 )
+
+                else
+                    ( 1, 1, midIfOutOfBounds (cfg.f10 / cfg.b) )
+        in
+        { edge0 = 0
+        , edge1 = edge1
+        , end00 = 0
+        , end01 = midIfOutOfBounds (cfg.f00 / cfg.b)
+        , end10 = end10
+        , end11 = end11
+        }
+
+    else if cfg.classify0 == 0 then
+        let
+            ( edge1, end10, end11 ) =
+                if cfg.classify1 < 0 then
+                    ( 0, 0, midIfOutOfBounds (cfg.f00 / cfg.b) )
+
+                else if cfg.classify1 == 0 then
+                    ( 3, cfg.sValue1, 1 )
+
+                else
+                    ( 1, 1, midIfOutOfBounds (cfg.f10 / cfg.b) )
+        in
+        { edge0 = 2
+        , edge1 = edge1
+        , end00 = cfg.sValue0
+        , end01 = 0
+        , end10 = end10
+        , end11 = end11
+        }
+
+    else
+        let
+            ( edge1, end10, end11 ) =
+                if cfg.classify1 == 0 then
+                    ( 3, cfg.sValue1, 1 )
+
+                else
+                    ( 0, 0, midIfOutOfBounds (cfg.f00 / cfg.b) )
+        in
+        { edge0 = 1
+        , edge1 = edge1
+        , end00 = 1
+        , end01 = midIfOutOfBounds (cfg.f10 / cfg.b)
+        , end10 = end10
+        , end11 = end11
+        }
+
+
+{-| Find the nearest points between two line segments. Returns `Nothing` if NaN or infinity would otherwise come up in the result.
+-}
+nearestPoints :
+    LineSegment3d u c
+    -> LineSegment3d u c
+    -> Maybe { nearestOnFirstLine : Point3d u c, nearestOnSecondLine : Point3d u c }
+nearestPoints p q =
+    -- Ported from: <https://www.geometrictools.com/Documentation/DistanceLine3Line3.pdf>
```

Comment:
Maybe this could live as a module comment, saying where the algorithm came from, that it's a port of code from that link. Also, I haven't checked the license

Reply:
I figured a module comment didn't make sense since other functions could get added to this module later which would be unrelated to this.

Reply:
Ah, that's true.